applications/CompressiblePotentialFlowApplication/custom_elements/compressible_potential_flow_element.h

```diff
@@ -527,11 +527,34 @@ class CompressiblePotentialFlowElement : public Element
 
                 const array_1d<double,Dim> v = prod(trans(data.DN_DX), data.phis);
 
-
                 p = (vinfinity_norm2 - inner_prod(v,v))/vinfinity_norm2; //0.5*(norm_2(vinfinity) - norm_2(v));
             }
+            else if(active==true && this->Is(MARKER))
+            {
+                const array_1d<double,3> vinfinity = rCurrentProcessInfo[VELOCITY];
```

Comment:
It may be worth it to define your own descriptive variable instead of using `VELOCITY`.